extra_data/tests/test_reader_mockdata.py

```diff
@@ -496,6 +496,16 @@ def test_select(mock_fxe_raw_run):
     for source, source_data in data.items():
         assert set(source_data.keys()) == {'image.pulseId', 'metadata'}
 
+    sel_by_list = run.select([
+        ('*/DET/*', 'image.pulseId'),
+        'FXE_XAD_GEC/CAM/*',
+    ])
+    assert 'SPB_XTD9_XGM/DOOCS/MAIN' not in sel_by_list.control_sources
+    assert 'FXE_DET_LPD1M-1/DET/0CH0:xtdf' in sel_by_list.instrument_sources
+    assert sel_by_list['FXE_DET_LPD1M-1/DET/0CH0:xtdf'].keys() == {'image.pulseId'}
+    assert 'FXE_XAD_GEC/CAM/CAMERA_NODATA' in sel_by_list.control_sources
+    assert 'FXE_XAD_GEC/CAM/CAMERA_NODATA:daqOutput' in sel_by_list.instrument_sources
+
```

Comment:
One might add a check here that all keys for one of the `FXE_XAD_GEC/CAM/*` is present.

Reply:
Good point, added a check.